### PR TITLE
test: add pytest openvino version check & add test_hello_classification_cpp_fail (invalid model)

### PR DESCRIPTION
**Pull request by dpigasin**

### CODE REVIEW

File: utils/arg_parser.py [hunk 1]
```diff
@@ -446,7 +447,20 @@ def parse_args(name: str, description: str):
             args.tags = [f'{args.os}_{args.distribution}:'
                          f'{args.build_id if args.build_id else args.product_version}',
                          f'{args.os}_{args.distribution}:latest']
+
     if args.mode not in ('test', 'deploy'):
         args.year = args.build_id[:4] if args.build_id else args.product_version[:4]
 
+    if args.mode == 'test' and not args.product_version:
+        match = re.search(r':(\d{4}\.\d\.\d{3})', str(args.tags))
```
Comment (generalova-kate) [inline utils/arg_parser.py:455]: tags can be YYYY.U without BBB

File: tests/functional/samples/test_samples_linux.py [hunk 2]
```diff
@@ -109,6 +109,33 @@ def test_hello_classification_cpp_hddl(self, is_distribution, is_image_os, teste
              ], self.test_hello_classification_cpp_hddl.__name__, **kwargs,
         )
 
+    @pytest.mark.parametrize('is_distribution', ['dev', 'proprietary'], indirect=True)
+    @pytest.mark.parametrize('is_image_os', ['ubuntu18', 'ubuntu20'], indirect=True)
+    @pytest.mark.parametrize('is_not_product_version', ['2020.3'], indirect=True)
+    @pytest.mark.xfail(reason='invalid model')
```
Comment (generalova-kate) [inline tests/functional/samples/test_samples_linux.py:115]: don't use xfail for really negative tests, you need catch raised exception FailedTest with exit code and check caplog with appropriate error text.

File: tests/functional/samples/test_samples_linux.py [hunk 3]
```diff
@@ -109,6 +109,33 @@ def test_hello_classification_cpp_hddl(self, is_distribution, is_image_os, teste
...
+    def test_hello_classification_cpp_fail(self, is_distribution, is_image_os, tester, image_os,
+                                           image, mount_root, is_not_product_version):
```
Comment (generalova-kate) [inline tests/functional/samples/test_samples_linux.py:117]: do we really need mount_root here?

File: tests/functional/samples/test_samples_linux.py [hunk 4]
```diff
@@ -109,6 +111,35 @@ def test_hello_classification_cpp_hddl(self, is_distribution, is_image_os, teste
              ], self.test_hello_classification_cpp_hddl.__name__, **kwargs,
         )
 
+    @pytest.mark.parametrize('is_distribution', ['dev', 'proprietary'], indirect=True)
+    @pytest.mark.parametrize('is_image_os', ['ubuntu18', 'ubuntu20'], indirect=True)
+    @pytest.mark.parametrize('is_not_product_version', ['2020.3'], indirect=True)
```
Comment (generalova-kate) [inline tests/functional/samples/test_samples_linux.py:116]: this test will be work only from 2021.2 release. I think your check (2020.3) is not full.

File: tests/conftest.py [hunk 5]
```diff
@@ -201,6 +201,13 @@ def is_package_url_specified(request):
         pytest.skip('Test requires a url for a dev package.')
 
 
+@pytest.fixture(scope='session')
+def min_product_version(request):
+    if request.param > request.config.getoption('--product_version'):
```
Comment (generalova-kate) [inline tests/conftest.py:206]: > or = (should be {request.param} or newer) it will not work in case 2021.2 and you have 'min_product_version', ['2021.2']

Comment (dpigasin) [reply tests/conftest.py:206]: It works in case '2021.2' because '2021.2' > '2021.2' = false

File: tests/conftest.py [hunk 6]
```diff
@@ -200,6 +201,26 @@ def is_package_url_specified(request):
...
+        pytest.skip(f'Test requires the product_version should be {request.param} or newer '
+                    f'but get {request.config.getoption("--product_version")}')
+
+
+@pytest.fixture(scope='session')
+def is_not_product_version(request):
```
Comment (generalova-kate) [inline tests/conftest.py:212]: let's remove it

**Review by generalova-kate — APPROVED** [review]

**Pull request merged**